lib/gears/shape.lua

```diff
@@ -168,6 +220,243 @@ function module.isosceles_triangle(cr, width, height)
     cr:close_path()
 end
 
+--- A cross (**+**) symbol.
+--
+-- ![Shape example](@SHAPE_cross_SVG@)
+--
+-- @SHAPE_cross_EXAMPLE@
+--
+-- @param cr A cairo context
+-- @tparam number width The shape with
+-- @tparam number height The shape height
+-- @tparam[opt=width/3] number thickness The cross section thickness
+function module.cross(cr, width, height, thickness)
+    local thickness = thickness or width/3
+    local xpadding   = (width  - thickness) / 2
+    local ypadding   = (height - thickness) / 2
+    cr:move_to(xpadding, 0)
+    cr:line_to(width - xpadding, 0)
+    cr:line_to(width - xpadding, ypadding)
+    cr:line_to(width           , ypadding)
+    cr:line_to(width           , height-ypadding)
+    cr:line_to(width - xpadding, height-ypadding)
+    cr:line_to(width - xpadding, height         )
+    cr:line_to(xpadding        , height         )
+    cr:line_to(xpadding        , height-ypadding)
+    cr:line_to(0               , height-ypadding)
+    cr:line_to(0               , ypadding       )
+    cr:line_to(xpadding        , ypadding       )
+    cr:close_path()
+end
+
+--- A similar shape to the `rounded_rect`, but with sharp corners.
+--
+-- ![Shape example](@SHAPE_octogon_SVG@)
+--
+-- @SHAPE_octogon_EXAMPLE@
+--
+-- @param cr A cairo context
+-- @tparam number width The shape with
+-- @tparam number height The shape height
+-- @tparam number corner_radius 
+function module.octogon(cr, width, height, corner_radius)
+    local corner_radius = corner_radius or math.min(10, math.min(width, height)/4)
+    local offset = math.sqrt( (corner_radius*corner_radius) / 2 )
+
+    cr:move_to(offset, 0)
+    cr:line_to(width-offset, 0)
+    cr:line_to(width, offset)
+    cr:line_to(width, height-offset)
+    cr:line_to(width-offset, height)
+    cr:line_to(offset, height)
+    cr:line_to(0, height-offset)
+    cr:line_to(0, offset)
+    cr:close_path()
+end
+
+--- A circle shape.
+--
+-- ![Shape example](@SHAPE_circle_SVG@)
+--
+-- @SHAPE_circle_EXAMPLE@
+--
+-- @param cr A cairo context
+-- @tparam number width The shape with
+-- @tparam number height The shape height
+function module.circle(cr, width, height)
+    local size = width < height and (width/2) or (height/2)
+    local w_offset, h_offset = (width - size*2)/2, (height - size*2)/2
+    cr:arc(w_offset + size, h_offset + size, size, 0, 2*math.pi)
+    cr:close_path()
+end
+
+--- A simple rectangle.
+--
+-- ![Shape example](@SHAPE_rectangle_SVG@)
+--
+-- @SHAPE_rectangle_EXAMPLE@
+--
+-- @param cr A cairo context
+-- @tparam number width The shape with
+-- @tparam number height The shape height
+function module.rectangle(cr, width, height)
+    cr:rectangle(0, 0, width, height)
+end
+
+--- A diagonal parallelogram with the bottom right corner at x=0 and top left
+-- at x=width.
```

Comment:
Isn't (0, 0) the top-left corner? I think this should say that the top-left corner is at x=0 and the bottom right one is at x=width. Also, you could mention that the top right corner is at (width-thickness, 0), that might simplify understanding this. (Is it usual to call this a thickness? I have no idea...)